Prep for init command

jnicklas opened the pull request:
This does a few things:

- Change the structure of how we load CLI options so we don't have to load project config for all commands. The init command will not want to load project config.
- Change format of command line args to the standard dasherized form, reduce weirdness tax
- Remove default app, instead we will provide a sensible default in the `init` command.

Closes #276

jnicklas committed:
Prep for init command

This does a few things:

- Change the structure of how we load CLI options so we don't have to load project config for all commands. The init command will not want to load project config.
- Change format of command line args to the standard dasherized form, reduce weirdness tax
- Remove default app, instead we will provide a sensible default in the `init` command.

jnicklas committed:
Better format for launch error

cowboyd commented on packages/project/src/index.ts:
```diff
@@ -47,16 +47,12 @@ export type ProjectOptions = {
   return {
     port: 24002,
-    app: defaultApp,
+    app: {
+      url: "about:blank"
+    },
```
Let's just allow the app service to error if no url is provided, and make sure that we initialize the boilerplate with one in our `init` command.

jnicklas replied:
Fixed!

cowboyd reviewed (changes requested):
See comment regarding default app options

jnicklas committed:
Hard failure when app url is not given

We assume that `app.url` is specified in the configuration. If it is not, we will throw an error.

A sensible default should be generated by the `init` command.
